monitoring/monitorlib/scd.py

```diff
@@ -431,3 +432,54 @@ def vol4s_intersect(vol4s_1: List[Volume4D], vol4s_2: List[Volume4D]) -> bool:
             if vol4_intersect(v1, v2):
                 return True
     return False
+
+
+def op_intent_transition_valid(
+    transition_from: Optional[OperationalIntentState],
+    transition_to: Optional[OperationalIntentState],
+) -> bool:
+    valid_states = {
+        OperationalIntentState.Accepted,
+        OperationalIntentState.Activated,
+        OperationalIntentState.Nonconforming,
+        OperationalIntentState.Contingent,
+    }
+    if (transition_from is not None and transition_from not in valid_states) or (
+        transition_to is not None and transition_to not in valid_states
+    ):
+        raise ValueError(
+            f"Either {transition_from} or {transition_to} is an invalid operational intent state"
+        )
+
+    if transition_from is None:
+        return transition_to == OperationalIntentState.Accepted
+
+    elif transition_from == OperationalIntentState.Accepted:
+        return transition_to in {
+            None,
+            OperationalIntentState.Accepted,
+            OperationalIntentState.Activated,
+            OperationalIntentState.Nonconforming,
+        }
```

Comment:
Although not shown in the diagram, Accepted can transition directly to Contingent.  See, e.g., 4.4.4.4:

> There are mutiple ways an operational intent can transition to the Contingent state, including manual initiation by UAS personnel (could occur from the Accepted, Activated, or Nonconforming states);
```suggestion
        return transition_to in {
            None,
            OperationalIntentState.Accepted,
            OperationalIntentState.Activated,
            OperationalIntentState.Nonconforming,
            OperationalIntentState.Contingent,
        }
```

Reply:
I had miss this, thanks.